dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs

```diff
@@ -55,10 +56,12 @@ public sealed class ActionPlanner : IActionPlanner
     /// <param name="kernel">The semantic kernel instance.</param>
     /// <param name="prompt">Optional prompt override</param>
     /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
+    /// <param name="config">The planner configuration.</param>
     public ActionPlanner(
         IKernel kernel,
         string? prompt = null,
-        ILoggerFactory? loggerFactory = null)
+        ILoggerFactory? loggerFactory = null,
+        ActionPlannerConfig? config = null)
```

Comment:
Maybe we can add `ActionPlannerConfig? config` parameter after `IKernel kernel` parameter to follow the same pattern as in `SequentialPlanner` and `StepwisePlanner` constructors.
I would definitely put `config` parameter before `ILoggerFactory`. I think having `ILoggerFactory` as last optional parameter is a good pattern and is used in most cases across codebase.

Reply:
I think we can move `prompt` to `config` as well. 
Regarding logging: having different logger factories for planner and kernel should provide flexibility, but I'm not sure it will be a common case, most probably users will want to configure logger once at kernel level and reuse it in all modules. We can also use provided in constructor `loggerFactory`, and in case it's `null` - use the one from Kernel. 

It can be done in separate PR, but in this one I would put `config` parameter after `kernel` for consistency.

Reply:
I changed the order of the args. 

Regarding putting the prompt in the config and using the logger factory of the kernel - it would make sense to make ActionPlanner more consistent with the others in general. For the prompt - the sequential planner uses a delegate for this like:
```
this._promptTemplate = this.Config.GetPromptTemplate?.Invoke() ?? EmbeddedResource.Read("Skills.StepwiseStep.skprompt.txt");
```
I think it'd make sense to match this pattern for the ActionPlanner. For the loggerFactory, I don't see a reason to be inconsistent with the other planners and provide a constructor arg when the others don't.

Either way, since there's questions about which one to go forward with - it would make sense to make a separate PR.